scripts/shared/clusters.sh

```diff
@@ -0,0 +1,116 @@
+    sed -i -- "s/server: .*/server: https:\/\/$master_ip:6443/g" $KUBECONFIG
+    sed -i -- "s/user: kind-.*/user: ${cluster}/g" $KUBECONFIG
+    sed -i -- "s/name: kind-.*/name: ${cluster}/g" $KUBECONFIG
+    sed -i -- "s/cluster: kind-.*/cluster: ${cluster}/g" $KUBECONFIG
+    sed -i -- "s/current-context: .*/current-context: ${cluster}/g" $KUBECONFIG
+    chmod a+r $KUBECONFIG
+}
+
+function create_kind_cluster() {
+    export KUBECONFIG=${KUBECONFIGS_DIR}/kind-config-${cluster}
+    if [[ $(kind get clusters | grep "^${cluster}$" | wc -l) -gt 0  ]]; then
+        echo "KIND cluster already exists, skipping its creation..."
+        kind export kubeconfig --name=${cluster}
+        kind_fixup_config ${cluster}
+        return
+    fi
+
+    echo "Creating KIND cluster..."
+    generate_cluster_yaml "${cluster}"
+    image_flag=''
+    if [[ -n ${version} ]]; then
+        image_flag="--image=kindest/node:v${version}"
+    fi
+
+    kind create cluster $image_flag --name=${cluster} --config=${RESOURCES_DIR}/${cluster}-config.yaml
+    kind_fixup_config ${cluster}
+}
+
+function deploy_weave_cni(){
+    if kubectl wait --for=condition=Ready pods -l name=weave-net -n kube-system --timeout=60s > /dev/null 2>&1; then
+        echo "Weave already deployed."
+        return
+    fi
+
+    echo "Applying weave network..."
+    kubectl apply -f "https://cloud.weave.works/k8s/net?k8s-version=v$version&env.IPALLOC_RANGE=${cluster_CIDRs[${cluster}]}"
+    echo "Waiting for weave-net pods to be ready..."
+    kubectl wait --for=condition=Ready pods -l name=weave-net -n kube-system --timeout=300s
+    echo "Waiting for core-dns deployment to be ready..."
+    kubectl -n kube-system rollout status deploy/coredns --timeout=300s
+}
+
+
+### Main ###
+
+LONGOPTS=k8s_version:,globalnet:
+# Only accept longopts, but must pass null shortopts or first param after "--" will be incorrectly used
+SHORTOPTS=""
+! PARSED=$(getopt --options=$SHORTOPTS --longoptions=$LONGOPTS --name "$0" -- "$@")
+eval set -- "$PARSED"
+
+while true; do
+    case "$1" in
+        --k8s_version)
+            version="$2"
+            ;;
+        --globalnet)
+            globalnet="$2"
+            ;;
+        --)
+            break
+            ;;
+        *)
+            echo "Ignoring unknown option: $1 $2"
+            ;;
+    esac
+    shift 2
+done
+
+echo "Running with: k8s_version=${version}, globalnet=${globalnet}"
+
+rm -rf ${KUBECONFIGS_DIR}
+mkdir -p ${KUBECONFIGS_DIR}
+
+# Run a local registry to avoid loading images manually to kind
+if registry_running; then
+    echo Local registry $KIND_REGISTRY already running.
+else
+    echo Deploying local registry $KIND_REGISTRY to serve images centrally.
+    docker run -d -p 5000:5000 --restart=always --name $KIND_REGISTRY registry:2
+fi
+
+# This IP is consumed by kind to point the registry mirror correctly to the local registry
+registry_ip="$(docker inspect -f '{{.NetworkSettings.IPAddress}}' "$KIND_REGISTRY")"
```

Comment:
We may want to extract this to a function in a different patch.

Reply:
Thanks, I'm doing this in another PR on top of this with some other cleanup work :)